File: core/src/SSRAG/Models/Segment.cs
using SSRAG.Datory;
using SSRAG.Datory.Annotations;

namespace SSRAG.Models
{
    [DataTable("ssrag_segment")]
    public class Segment : Entity
    {
        [DataColumn]
        public int SiteId { get; set; }

        [DataColumn]
        public int ChannelId { get; set; }

        [DataColumn]
        public int ContentId { get; set; }

        [DataColumn]
        public int DocumentId { get; set; }

        [DataColumn]
        public int Taxis { get; set; }

        [DataColumn(Text = true)]
        public string Text { get; set; }

        [DataColumn]
        public string TextHash { get; set; }
    }
}

File: ai/models/segment.py
from sqlmodel import Field, SQLModel, Column, Integer, String, DateTime, Text
from datetime import datetime
from configs import table_names


class Segment(SQLModel, table=True):
    __tablename__ = table_names.TABLE_NAME_SEGMENT  # type: ignore
    id: str = Field(
        sa_column=Column("Id", String(50), primary_key=True),
        default=None,
    )
    createdDate: datetime = Field(
        sa_column=Column("CreatedDate", DateTime, default=datetime.now),
        default=None,
    )
    lastModifiedDate: datetime = Field(
        sa_column=Column("LastModifiedDate", DateTime, default=datetime.now),
        default=None,
    )
    siteId: int = Field(
        sa_column=Column("SiteId", Integer),
        default=None,
    )
    channelId: int = Field(
        sa_column=Column("ChannelId", Integer),
        default=None,
    )
    contentId: int = Field(
        sa_column=Column("ContentId", Integer),
        default=None,
    )
    documentId: int = Field(
        sa_column=Column("DocumentId", Integer),
        default=None,
    )
    taxis: int = Field(
        sa_column=Column("Taxis", Integer),
        default=None,
    )
    text: str = Field(
        sa_column=Column("Text", Text),
        default=None,
    )
    textHash: str = Field(
        sa_column=Column("TextHash", String(500)),
        default=None,
    )
